orga/app/routes/authenticated/campaigns/participant-assessment.js

```diff
@@ -1,15 +1,22 @@
 import Route from '@ember/routing/route';
 import RSVP from 'rsvp';
+import { inject as service } from '@ember/service';
 
 export default class AssessmentRoute extends Route {
-  model(params) {
-    const { campaign_id: campaignId, campaign_participation_id: campaignParticipationId } = params;
-    return RSVP.hash({
-      campaign: this.store.findRecord('campaign', campaignId),
-      campaignAssessmentParticipation: this.store.queryRecord('campaign-assessment-participation', {
-        campaignId,
-        campaignParticipationId,
-      }),
-    });
+  @service router;
+
+  async model(params) {
+    try {
+      const { campaign_id: campaignId, campaign_participation_id: campaignParticipationId } = params;
+      return await RSVP.hash({
+        campaign: this.store.findRecord('campaign', campaignId),
+        campaignAssessmentParticipation: this.store.queryRecord('campaign-assessment-participation', {
+          campaignId,
+          campaignParticipationId,
+        }),
+      });
+    } catch (error) {
```

Comment:
**question** : quelque chose avec la gestion d'événement erreur de ember nativement ça marche pas ? https://guides.emberjs.com/release/routing/loading-and-error-substates/#toc_the-error-event

Reply:
Le return await avec deux promises  dans le try je le trouve un peu difficile à suivre.

Reply:
ah bonne question. j'avais vu une route faite comme ça. j'ai fait pareil 😅 . Je peux regarder

Reply:
j'ai tenté. il y a apparement un bug ember ou une mauvaise gestion de notre part des pages not-found. J'ai l'impression qu'à moins de revoir notre système de redirection d'erreur / page non trouvé.

https://github.com/emberjs/ember.js/issues/18746